postgres/datadog_checks/postgres/statement_samples.py

```diff
@@ -486,6 +575,52 @@ def _explain_pg_stat_activity(self, rows):
                     tags=self._tags + ["error:collect-plan-for-statement-crash"] + self._check._get_debug_tags(),
                     hostname=self._check.resolved_hostname,
                 )
+        return events
+
+    def _create_activity_event(self, rows, active_connections):
+        self._time_since_last_activity_event = time.time()
+        active_sessions = self._to_active_sessions(rows)
+        if len(active_sessions) > self._activity_max_rows:
+            self._check.gauge(
+                "dd.postgres.create_activity_event.active_sessions_pre_truncate.size",
+                len(active_sessions),
+                tags=self._tags + self._check._get_debug_tags(),
+                hostname=self._check.resolved_hostname,
+            )
+            active_sessions = self._truncate_activity_rows(active_sessions)
+        event = {
+            "host": self._db_hostname,
+            "ddagentversion": datadog_agent.get_version(),
+            "ddsource": "postgres",
+            "dbm_type": "activity",
+            "collection_interval": self._activity_coll_interval,
+            "ddtags": self._tags_no_db,
+            "timestamp": time.time() * 1000,
+            "postgres_activity": active_sessions,
+            "postgres_connections": active_connections,
+        }
+        return event
+
+    def _truncate_activity_rows(self, rows):
```

Comment:
`_truncate_activity_rows(self, rows, limit)` to make it easy to unit test directly